File: wr-app/src/api/models/erp/ammunition-consumption.ts
export interface AmmunitionStockOption {
    balanceId: number; lotId: number; sku: string; modelName: string; lotNumber: string; locationName: string;
    unitOfMeasure: string; available: number; validUntil: string | null;
}
export interface AmmunitionConsumptionItem extends AmmunitionStockOption {
    id: number; quantity: number; result: string; movementId: number;
}
export interface AmmunitionConsumption {
    id: number; organizationId: number; organizationName: string; unitId: number | null; unitName: string | null;
    responsibleId: number; responsibleName: string; authorizerId: number; authorizerName: string; purpose: string;
    status: string; consumedAt: string; finalizedById: number | null; finalizedByLogin: string | null;
    items: AmmunitionConsumptionItem[];
}
export interface AmmunitionConsumptionRequest {
    requestId: string; organizationId: number; unitId?: number; responsibleId: number; authorizerId: number;
    purpose: string; items: { balanceId: number; quantity: number; result: string }[];
}
export interface AmmunitionConsumptionPage<T> { content: T[]; totalElements: number; page: number; size: number }

File: wr-api/src/main/java/com/weaponsregistration/consumption/model/AmmunitionConsumption.java
package com.weaponsregistration.consumption.model;

import com.weaponsregistration.core.model.*;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "erp_ammunition_consumption")
public class AmmunitionConsumption extends CoreEntity {
    @ManyToOne(optional = false) @JoinColumn(name = "organization_id", nullable = false) public Organization organization;
    @ManyToOne @JoinColumn(name = "unit_id") public OrganizationalUnit unit;
    @ManyToOne(optional = false) @JoinColumn(name = "responsible_id", nullable = false) public Person responsible;
    @ManyToOne(optional = false) @JoinColumn(name = "authorizer_id", nullable = false) public Person authorizer;
    @Column(nullable = false) public String organizationName;
    @Column public String unitName;
    @Column(nullable = false) public String responsibleName;
    @Column(nullable = false) public String authorizerName;
    @Column(nullable = false, length = 255) public String purpose;
    @Column(name = "activity_type", nullable = false, length = 40) public String activityType = "OPERATION";
    @Column(name = "operation_training", length = 255) public String operationTraining;
    @Column(nullable = false, length = 30) public String status = "FINALIZED";
    @Column(nullable = false) public LocalDateTime consumedAt;
    @Column public Long finalizedById;
    @Column public String finalizedByLogin;
    @Column(nullable = false, unique = true, length = 36) public String requestId;
    @Column(nullable = false, length = 64) public String requestFingerprint;
}
